### PR TITLE
Replace ParsedDeclaration::expand with non-generic method

### PR DESCRIPTION
**Pull request by SimonSapin**
... to reduce its code size impact.

https://bugzilla.mozilla.org/show_bug.cgi?id=1351737

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/16183)

### CODE REVIEW

File: components/style/properties/properties.mako.rs [hunk 1]
```diff
@@ -899,31 +921,57 @@ impl ParsedDeclaration {
...
+                                % if sub_property.boxed:
+                                    Box::new(${sub_property.ident})
+                                % else:
+                                    ${sub_property.ident}
+                                % endif
+                            ),
+                            importance,
+                            overwrite_more_important,
+                        )
+                        ||
```
Comment (emilio) [inline components/style/properties/properties.mako.rs]: This can't short-circuit.

File: components/style/properties/properties.mako.rs [hunk 2]
```diff
@@ -899,31 +921,57 @@ impl ParsedDeclaration {
...
-                        push(PropertyDeclaration::CSSWideKeyword(LonghandId::${sub_property.camel_case}, keyword));
+                        block.push_common(
+                            PropertyDeclaration::CSSWideKeyword(
+                                LonghandId::${sub_property.camel_case},
+                                keyword,
+                            ),
+                            importance,
+                            overwrite_more_important,
+                        )
+                        ||
```
Comment (emilio) [inline components/style/properties/properties.mako.rs]: Nor this.

**Review by emilio — CHANGES_REQUESTED** [review]
r=me with those fixed

**Comment by Manishearth** [comment]
er.

File: components/style/properties/properties.mako.rs (hunk 1 above)
Comment (Manishearth) [inline components/style/properties/properties.mako.rs]: Do we short circuit things with side effects?

Comment (Manishearth) [inline components/style/properties/properties.mako.rs]: Huh, we do. Interesting.

Comment (SimonSapin) [inline components/style/properties/properties.mako.rs]: The `||` operator doesn’t know whether its operands have side effects.

**Comment by SimonSapin** [comment]
```
  ▶ CRASH [expected TIMEOUT] /html/semantics/embedded-content/the-iframe-element/iframe_sandbox_popups_nonescaping-1.html
  │ 
  │ VMware, Inc.
  │ Gallium 0.4 on softpipe
  │ 3.3 (Core Profile) Mesa 12.0.1
  │ ERROR:script::dom::bindings::error: Error at http://web-platform.test:8000/html/semantics/embedded-content/the-iframe-element/iframe_sandbox_popups_helper-1.html:3:1 opener is not defined
  │ called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }) } (thread RenderBackend, at /checkout/src/libcore/result.rs:859)
  │ stack backtrace:
  │    0:     0x7f7f04f0f5fd - backtrace::backtrace::trace::h43650c97fbd1994d
  │    1:     0x7f7f04f0fc02 - backtrace::capture::Backtrace::new::hb5a725a088a2a2fc
  │    2:     0x7f7f047f33d7 - servo::main::{{closure}}::h9d96b5c32a8afffa
  │    3:     0x7f7f0675a13d - std::panicking::rust_panic_with_hook
  │                         at /checkout/src/libstd/panicking.rs:550
  │    4:     0x7f7f06759fd4 - std::panicking::begin_panic<collections::string::String>
  │                         at /checkout/src/libstd/panicking.rs:511
  │    5:     0x7f7f06759f09 - std::panicking::begin_panic_fmt
  │                         at /checkout/src/libstd/panicking.rs:495
  │    6:     0x7f7f06759e97 - std::panicking::rust_begin_panic
  │                         at /checkout/src/libstd/panicking.rs:471
  │    7:     0x7f7f0678669d - core::panicking::panic_fmt
  │                         at /checkout/src/libcore/panicking.rs:69
  │    8:     0x7f7f04c21d39 - core::result::unwrap_failed::h6bfb389c41db9e3b
  │    9:     0x7f7f04c8ee5d - webrender::render_backend::RenderBackend::run::hfa58adaca88f10b0
  │   10:     0x7f7f04c01cff - std::panicking::try::do_call::h17ab1154c25f8791
  │   11:     0x7f7f0676100a - panic_unwind::__rust_maybe_catch_panic
  │                         at /checkout/src/libpanic_unwind/lib.rs:98
  │   12:     0x7f7f04c22ef4 - <F as alloc::boxed::FnBox<A>>::call_box::h87f7ba67e73da805
  │   13:     0x7f7f06758c44 - alloc::boxed::{{impl}}::call_once<(),()>
  │                         at /checkout/src/liballoc/boxed.rs:650
  │                          - std::sys_common::thread::start_thread
  │                         at /checkout/src/libstd/sys_common/thread.rs:21
  │                          - std::sys::imp::thread::{{impl}}::new::thread_start
  │                         at /checkout/src/libstd/sys/unix/thread.rs:84
  │   14:     0x7f7f026be183 - start_thread
  │   15:     0x7f7f021d5bec - clone
  │   16:                0x0 - <unknown>
  │ ERROR:servo: called `Result::unwrap()` on an `Err` value: Error { repr: Custom(Custom { kind: UnexpectedEof, error: StringError("failed to fill whole buffer") }) }
  └ Pipeline failed in hard-fail mode.  Crashing!
```

@bors-servo retry #14323

**Pull request merged**